src/S3Restore.php

```diff
@@ -204,27 +204,28 @@ function ($bucket) {
             }
 
             // upload data
-            $slices = $this->s3Client->listObjects(
-                [
-                    'Bucket' => $sourceBucket,
-                    'Prefix' => $sourceBasePath . str_replace('.', '/', $tableId) . '.',
-                ]
-            );
+            $iterator = $this->s3Client->getIterator('ListObjects', [
+                'Bucket' => $sourceBucket,
+                'Prefix' => $sourceBasePath . str_replace('.', '/', $tableId) . '.',
+            ]);
+
+            $slices = iterator_to_array($iterator);
 
             // no files for the table found, probably an empty table
-            if (!isset($slices["Contents"])) {
+            if (count($slices) === 0) {
                 unset($headerFile);
                 continue;
             }
 
-            if (count($slices["Contents"]) === 1 && substr($slices["Contents"][0]["Key"], -14) !== '.part_0.csv.gz') {
+            $firstSlice = reset($slices);
+            if (count($slices) === 1 && substr($firstSlice["Key"], -14) !== '.part_0.csv.gz') {
```

Comment:
Toto je trosku nejasne, to cislo `-14`  by som radsej videl ako `-strlen('.part_0.csv.gz')` alebo tak ...

```php
substr($firstSlice["Key"], -14) !== '.part_0.csv.gz'
```
... ale pozeram, ze tot ak bolo aj predtym.

Reply:
Jj to je prevzate. Bere to 14 znaku od konce. Cili jeslti  nazev souboru konci na `.part_0.csv.gz`